lib/rules/no-warning-comments.js

```diff
@@ -141,12 +143,29 @@ module.exports = {
 
             const matches = commentContainsWarningTerm(node.value);
 
-            matches.forEach(matchedTerm => {
+            matches.forEach(({ type, comment }) => {
+                let len = 0;
+                const commentsToDisplay = [];
```

Comment:
Thoughts about building the final string instead of an array? That way we could avoid one extra variable `len`.